src/deepsparse/transformers/pipelines_cli.py

```diff
@@ -0,0 +1,260 @@
+        type=str,
+        default="sentiment-analysis",
+    )
+
+    parser.add_argument(
+        "-d",
+        "--data",
+        help="Path to file containing data for inferences, "
+        "inputs should be separated via newline",
+        required=True,
+        type=str,
+    )
+
+    parser.add_argument(
+        "--model-name",
+        "--model_name",
+        help="Canonical name of the hugging face model this model is based on",
+        default=None,
+        type=Optional[str],
+    )
+
+    parser.add_argument(
+        "--model-path",
+        "--model_path",
+        help="Path to (ONNX) model file to run, can also be a SparseZoo stub",
+        required=True,
+        type=str,
+    )
+
+    parser.add_argument(
+        "--engine-type",
+        "--engine_type",
+        help="Inference engine name to use. Supported options are 'deepsparse'"
+        "and 'onnxruntime'",
+        type=str,
+        choices=SUPPORTED_ENGINES,
+        default=SUPPORTED_ENGINES[0],
+    )
+
+    parser.add_argument(
+        "--config",
+        help="Huggingface model config, if none provided, default will be used"
+        "which will be from the model name or sparsezoo stub if given for "
+        "model path",
+        type=str,
+        default=None,
+    )
+
+    parser.add_argument(
+        "--tokenizer",
+        help="Huggingface tokenizer, if none provided, default will be used",
+        type=str,
+        default=None,
+    )
+
+    parser.add_argument(
+        "--max-length",
+        "--max_length",
+        help="Maximum sequence length of model inputs. default is 128",
+        type=int,
+        default=128,
+    )
+
+    parser.add_argument(
+        "--num-cores",
+        "--num_cores",
+        help="Number of CPU cores to run engine with. Default is the maximum "
+        "available",
+        type=int,
+        default=None,
+    )
+    parser.add_argument(
+        "-b",
+        "--batch-size",
+        "--batch_size",
+        help="The batch size to use for pipelines",
+        type=int,
+        default=1,
+    )
+
+    scheduler_choices = ["multi", "single"]
+    parser.add_argument(
+        "--scheduler",
+        choices=scheduler_choices,
+        help="The scheduler to use for the engine. Can be None, single or multi",
+        default=scheduler_choices[0],
+    )
+
+    parser.add_argument(
+        "-o",
+        "--output-file",
+        "--output_file",
+        action="store",
+        dest="output_file",
+        type=argparse.FileType("w"),
+        help="Directs the output to a name of your choice",
+        default="-",
+    )
+
+    _args = parser.parse_args()
+    data_file = Path(_args.data)
+    assert data_file.exists(), f"The input data file must exist, {data_file}"
+    assert data_file.is_file(), f"The input data file must be a file, {data_file}"
+    supported_extensions = [".json", ".csv", ".txt"]
+    assert data_file.suffix in supported_extensions, (
```

Comment:
let's rewrite these to raise specific exceptions

Reply:
Done!